### PR TITLE
Pause/resume node implemented

### LINKED ISSUE #96: Pause node
- in order to preserve resources we need to completely stop the underlying Geth node
- there's a twist: we need to make sure that whatever account was selected when node was paused, gets re-selected when node resumes (including adding that account into Whisper)

### PR DESCRIPTION
**Pull request by farazdagi**
- fixes #96
- merge https://github.com/status-im/status-go/pull/91 and https://github.com/status-im/status-go/pull/95 first (as this implementation depends on refactored implementation of transaction queue - only that updated implementation works correctly on stop/start events)
- new methods exposed: `StopNode()` and `ResumeNode()` (additional method for resuming is introduced due to fact that you do not need to provide `datadir` on resuming, so I've decided not implement resuming as part of `StartNode()`)
- both exposed methods and internal implementation are covered with tests (I made sure I double check  Whisper identity on resume)

**Note:**
- on `StopNode()` **all** Geth protocols stop (LES, IPC, HTTP etc)
- on `ResumeNode()` Geth protocols start again, previously selected ("logged in") account is injected into Whisper as identity

### CODE REVIEW

**Comment by farazdagi** [comment]
I've enabled WS-RPC server support. Couple of notes:
- Geth client is cool enough to connect to either HTTP, WS (websocket), or IPC. So, to test that WS-RPC is running all you need is to `geth attach ws://localhost:8546` (HTTP-RPC is still available on 8545).
- React Native seems to [support Websocket](https://facebook.github.io/react-native/docs/network.html) and with mere 150 LOCs we can have [reconnecting websockets](https://github.com/pladaria/reconnecting-websocket). So, **theoretically** this can be a solution for our problem with stalling socket on Android. Btw, that reconnecting websocket's reconnection is claimed to follow [RFC6455/7.2.3 Recovering from Abnormal Closure](https://tools.ietf.org/html/rfc6455#section-7.2.3) guidelines.
- When it comes to web3.js then I am not sure that it has WebSocket support (we might need to write provider for that, which doesn't seem to be too complex). However, it has [IPC provider](https://github.com/ethereum/web3.js/blob/develop/lib/web3/ipcprovider.js), so it seems IPC is supported natively. Another option is [web3_ipc node module](https://www.npmjs.com/package/web3_ipc). 

To sum up, Geth can be accessed via HTTP, WS or IPC. All of those protocols are restarted on node pause/resume (and pass all the tests we have). All three of those protocols can be used.

**Comment by rasom** [comment]
> When it comes to web3.js then I am not sure that it has WebSocket support ...

https://github.com/ethereum/web3.js/blob/1.0/lib/web3/providers/websocketprovider.js

Thanks, @farazdagi , I will try to use WS provider soon.

**Comment by jarradh** [comment]
IPC is the best, I didn't see how this can be easily done on android & ios though

**Comment by farazdagi** [comment]
@rasom can you please review this PR: pause/resume node should work as expected (including reselecting previously selected account). When it comes to IPC and WS servers, that functionality is disabled by default (but I still want to include it - as it mainly for optionally configuring node, and I'd rather not re-code that configuration should we need that for some future fixes).

**Comment by rasom** [comment]
Ok, thanks.
